### ISSUE TITLE
Improve naming of diff view's right hand side

### ISSUE THREAD

**Issue by weinand** [microsoft/vscode#104072]
Testing #103613

After turning on Settings Sync on a second system, I got a merge conflict and opened the diff editor.

<img width="402" alt="2020-08-05_15-20-38" src="https://user-images.githubusercontent.com/1898161/89417681-49cfa080-d72f-11ea-9be4-466abdf9f770.png">

In the diff editor the right hand side's label "settings.json (Merges)" is confusing because it says "Merges" but there is nothing merged initially.
The right hand side show the "local" settings.
Only after editing they become the "Merged local settings".

Could we say initially "settings.json (Local)" and after editing "settings.json (Merged local)"?

If not, it would be good if a native speaker could verify that the "Merges" (used both in the title and the button inside the editor) is correct English.

**Comment by sandy081** [comment]
> In the diff editor the right hand side's label "settings.json (Merges)" is confusing because it says "Merges" but there is nothing merged initially.

Right hand side do contain merges which are done automatically from local and remote and the diff shows those cannot be merged automatically. I think in current insiders user see this without performing any merge action which is confusing. I changed this experience today as follows:

- Turn on Sync on a second machine
- You get merge/replace dialog

<img width="650" alt="image" src="https://user-images.githubusercontent.com/10746682/89427920-147d7f80-d73c-11ea-8f9c-b589f64b6e6b.png">

- You choose to merge
- If there are conflicts, following dialog is shown asking user to resolve conflicts manually

<img width="458" alt="image" src="https://user-images.githubusercontent.com/10746682/89427969-23fcc880-d73c-11ea-88ce-39b2244cb64e.png">

- This opens Manual Merge view allowing user to merge manually where user has options to replace local or remote or merge and resolve conflicts.

<img width="257" alt="image" src="https://user-images.githubusercontent.com/10746682/89428102-50184980-d73c-11ea-9d2d-59cbfa20bda7.png">

- Conflicts editor (ie., in description) is shown only after user select merge action and these labels make sense given that user tried to merge.

There is also another issue opened already for handling conflicts UX - https://github.com/microsoft/vscode/issues/103950. 
So I would like to make this dup of that and continue discussion there to improve the UX.
